app/views/layouts/_discussions.html.erb

```diff
@@ -42,22 +42,22 @@
   <% else %>
     <div class="discussions">
       <% @filtered_discussions.each do |discussion| %>
-        <%= link_to item_discussion_path(discussion) do %>
+        <%= link_to exercise_discussion_path(discussion.exercise.id, discussion) do %>
           <div class="discussion">
             <div class="discussion-row">
               <%= discussion_messages_icon(discussion) %>
               <% unless @debatable.respond_to? :language %>
                 <div class="discussion-language-icon">
-                  <%= language_icon(discussion.language) %>
+                  <%= language_icon(discussion.exercise.language) %>
                 </div>
               <% end %>
               <div>
                 <div class="discussion-title">
                   <span class="discussion-status-icon">
                     <%= discussion_status_fa_icon(discussion) %>
                   </span>
-                  <span class="hidden-sm hidden-xs"><%= discussion.exercise.friendly %></span>
-                  <span class="hidden-md hidden-lg"><%= discussion.exercise.name %></span>
+                  <span class="hidden-sm hidden-xs"><%= discussion.exercise.guide.name %> -</span>
```

Comment:
Same as: https://github.com/mumuki/mumuki-laboratory/pull/1451/files#r466562227.
This has a downside: The full name of an exercise won't be displayed, just guide name and exercise name.

Reply:
I am normally against deep nested message chains, but if you think it is the best... :shrug: